tests/PHPStan/Analyser/nsrt/integer-range-types.php

```diff
@@ -446,3 +446,36 @@ public function zeroIssues($positive, $negative)
 	}
 
 }
+
+function subtract($m) {
+	if ($m != 0) {
+		assertType("mixed", $m); // could be "mixed~0|0.0|''|'0'|array{}|false|null"
+		assertType('int', (int) $m); // could be int<min, -1>|int<1, max>
+	}
+	if ($m !== 0) {
+		assertType("mixed~0", $m);
+		assertType('int', (int) $m); // mixed could still contain falsey values, which cast to 0
+	}
+	if (!is_int($m)) {
+		assertType("mixed~int", $m);
+		assertType('int', (int) $m); // mixed could still contain falsey values, which cast to 0
+	}
+
+	if ($m != true) {
+		assertType("0|0.0|''|'0'|array{}|false|null", $m);
+		assertType('0', (int) $m);
+	}
+	if ($m !== true) {
+		assertType("mixed~true", $m);
+		assertType('int', (int) $m);
+	}
+
+	if ($m != false) {
+		assertType("mixed~0|0.0|''|'0'|array{}|false|null", $m);
+		assertType('int<min, -1>|int<1, max>', (int) $m);
```

Comment:
running [my new tool](https://x.com/markusstaab/status/1834140685059977219) thru this test, told me that `(int) $m` can still be zero at this point:

https://3v4l.org/AiaKX